docs/book/src/reference/distributing-projects.md

```diff
@@ -0,0 +1,244 @@
+    ├── crds/
+    ├── manifests/
+    └── samples/
+  /Makefile
+  /PROJECT
+  /Dockerfile
+  ...
+  ```
+
+- **Kustomize**: Kubebuilder uses Kustomize to manage Kubernetes manifests, which simplifies the customization and deployment process.
+- **Operator Versions**: Each Operator release is versioned (e.g., 0.1.0), and you can use release channels such as alpha and stable to manage versions.
+- **Channels**: By default, the alpha channel is used for experimental or preview versions. You can configure this to use other channels based on your needs.
+
+### Step-by-Step Guide: Publishing Kubebuilder Projects to OperatorHub.io
+
+## Step 1: Prepare Your Kubebuilder Project
+
+Ensure that your Kubebuilder project is properly created and contains the necessary Kubernetes manifests and CRD files.
+
+### Step 1: To initialize a basic Kubebuilder project
+
+```
+kubebuilder init --domain mydomain.com --repo github.com/myorg/my-operator
+```
+
+### Step 2: Create your API resources
+
+```
+kubebuilder create api --group app --version v1 --kind MyApp
+```
+This will create the necessary directories and files under the config/ directory.
+
+### Step 3: Compltete your business and create your crds
+
+```
+make manifests
+```
+
+## Step 2: Generate the Operator Bundle
+
+### Step 1:  operator-sdk generate kustomize manifests
+
+Example output:
+```
+    operator-sdk generate kustomize manifests
+    Generating kustomize files in config/manifests
+
+    Display name for the operator (required):
+    > my-operator
+
+    Description for the operator (required):
+    > To verify how to generate the bundle
+
+    Provider's name for the operator (required):
+    > zhilongwang
+
+    Any relevant URL for the provider name (optional):
+    > https://github.com/rayowang
+
+    Comma-separated list of keywords for your operator (required):
+    > crd,custom-resource,controller
+
+    Comma-separated list of maintainers and their emails (e.g. 'name1:email1, name2:email2') (required):
+    > rayowang:rayo.wangzl@gmail.com
+    Kustomize files generated successfully
+```
+This generates the file: config/manifests/bases/kubebuilderdemo.clusterserviceversion.yaml.
+
+### Step 2: Add the config/manifests/kustomization.yaml
+
+These resources constitute the fully configured set of manifests, used to generate the 'manifests/' directory in a bundle.
+
+```
+resources:
+- bases/kubebuilderdemo.clusterserviceversion.yaml
+- ../default
+- ../samples
+```
+
+### Step 3: Generate the Operator Bundle
+
+Operator bundles are the standard format for distributing Operators. These bundles include the necessary Kubernetes manifests, CRDs, and metadata required by platforms like OperatorHub.io and OLM.
+
+Note: Due to compatibility issues between the latest Kubebuilder v4 and Operator SDK, we need to disable the cliVersion field before running the command. To do so, execute:
+
+```
+# On macOS
+sed -i '' '/^cliVersion:/s/^/#/' PROJECT
+# On Linux
+sed -i '/^cliVersion:/s/^/#/' PROJECT
```

Comment:
We do not need add it in the default doc it is a temp workaround.
It ONLY occurs if we built the solution with Kubebuilder 4.6.0 ( not previous ) and soon it will not be required.